source/neuropod/internal/backend_registration.cc

```diff
@@ -51,44 +53,74 @@ void init_registrar_if_needed()
     });
 }
 
-// A map from a backend type (e.g. tensorflow, python, torchscript, etc.) and version to
-// the name of a shared library that supports that type. These will only be loaded if a backend
-// for the requested type hasn't already been loaded.
+std::vector<BackendLoadSpec> get_default_backend_map()
+{
+    std::vector<BackendLoadSpec> out;
+
+    // A structure to store some basic info about frameworks we support
+    struct FrameworkInfo
+    {
+        std::string              type;
+        std::string              soname;
+        bool                     has_gpu_version;
+        std::vector<std::string> versions;
+    };
+
+    // Information about frameworks that we use to generate a list of `BackendLoadSpec`
+    const std::vector<FrameworkInfo> frameworks = {
+        {"torchscript", "libneuropod_torchscript_backend.so", true, {"1.1.0", "1.2.0", "1.3.0", "1.4.0", "1.5.0"}},
+        {"tensorflow", "libneuropod_tensorflow_backend.so", true, {"1.12.0", "1.13.1", "1.14.0", "1.15.0"}},
+        {"python", "libneuropod_pythonbridge_backend.so", false, {"27", "35", "36", "37", "38"}}};
+
+    // Because the returned vector is in reverse priority order,
+    // the ordering below means we'd prefer to load a newer, GPU capable version of a framework if one is available.
+    // It also prioritizes non-absoltute so paths (i.e. controlled by LD_LIBRARY_PATH) over absolute ones in
+    // `/usr/local/lib/neuropod`. This is so we don't break existing behavior.
+    for (const auto &is_absolute_path : {false, true})
+    {
+        for (const auto &is_gpu : {false, true})
+        {
+            for (const auto &framework : frameworks)
+            {
+                if (is_gpu && !framework.has_gpu_version)
+                {
+                    // This framework doesn't have a GPU specific version
+                    continue;
+                }
+
+                for (const auto &version : framework.versions)
+                {
+                    BackendLoadSpec item;
+                    item.type    = framework.type;
+                    item.version = version;
+
+                    if (is_absolute_path)
+                    {
+                        // Ex:
```

Comment:
Is it worth validating the existence of the file if is_absolute_path is true, and skip appending to 'out' if we find it doesn't exist?

Reply:
Probably not for the following reasons:
- If a file exists, it doesn't necessarily mean `dlopen` will work
- For relative paths, internally `dlopen` effectively just tries a bunch of absolute paths based on the system loader paths and `LD_LIBRARY_PATH`. Therefore, the approach here of trying several candidate locations likely isn't going to be "expensive"